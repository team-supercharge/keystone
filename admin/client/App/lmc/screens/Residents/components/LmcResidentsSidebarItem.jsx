import React from 'react'
import PropTypes from 'prop-types'

const PLACEHOLDER_IMAGE = 'https://s3-eu-west-2.amazonaws.com/lmc-marketing-public/wp-content/uploads/2018/04/12092141/profile_pic_placeholder.png';

const LmcResidentsSidebarItem = ({ resident, onClick, isSelected }) => {
    const img = Object.entries(resident.fields.picture).length ? resident.fields.picture : PLACEHOLDER_IMAGE
    const chosenImgStyle = isSelected ? styles.selectedImg : styles.regularImg
    const chosenStyle = isSelected ? styles.selected : styles.regular
    const chosenTextStyle = isSelected ? styles.selectedText : styles.regularText
    const imgBackground = {
        background: `url(${img})`
    }

    return (
        <li
            id={resident.id}
            onClick={onClick}
            style={{ ...styles.item, ...chosenStyle }}
        >
            <div>
                <span className='lmc-profile-picture__small' style={{ ...chosenImgStyle, ...styles.image, ...imgBackground }} />                
                <span style={{ ...chosenTextStyle, ...styles.nameText }}>{resident.name}</span>
            </div>
        </li>
    )
}

const styles = {
    image: {
        float: 'left',
        marginTop: '4px',
    },
    item : {
        listStyleType: 'none',
        height: '50px',
        display: 'flex',
		padding: '3px 0 3px 6px',
    },
    nameText: {
        marginLeft: '15px',
        position: 'relative',
        top: '12px',
        cursor: 'pointer',
    },
    selected: {
        background: '#f1f1f1',
		fontSize: 14,
		boxSizing: 'border-box',
		borderLeft: '3px solid #e65d78',
    },
    selectedImg: {
        marginLeft: '3px',
    },
    selectedText: {
        color: '#e65d78',
    },
    regular: {
        background: '#fafafa',
    },
    regularImg: {
        marginLeft: '6px',
    }
}

LmcResidentsSidebarItem.propTypes = {
    resident: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired,
    isSelected: PropTypes.bool.isRequired,
}

export default LmcResidentsSidebarItem