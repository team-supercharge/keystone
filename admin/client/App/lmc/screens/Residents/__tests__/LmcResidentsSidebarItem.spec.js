import React from 'react'
import { shallow } from 'enzyme'
import LmcResidentsSidebarItem from '../components/LmcResidentsSidebarItem'

describe('LmcResidentsSidebarItem', () => {
    let wrapper
    let resident

    beforeEach(() => {
        resident = {
            id: 'testId',
            name: 'testName',
            fields: {
                picture: 'testPicture',
            }
        }
        wrapper = shallow(
            <LmcResidentsSidebarItem
                resident={resident}
                onClick={() => {}}
                isSelected={true}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders a list item', () => {
        expect(wrapper.find('li').length).toEqual(1)
    })
})