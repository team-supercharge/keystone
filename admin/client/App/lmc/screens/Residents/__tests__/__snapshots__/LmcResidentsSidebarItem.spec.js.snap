// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcResidentsSidebarItem renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    id="testId"
    onClick={[Function]}
    style={
      Object {
        "background": "#f1f1f1",
        "borderLeft": "3px solid #e65d78",
        "boxSizing": "border-box",
        "display": "flex",
        "fontSize": 14,
        "height": "50px",
        "listStyleType": "none",
        "padding": "3px 0 3px 6px",
      }
    }
  >
    <div>
      <span
        className="lmc-profile-picture__small"
        style={
          Object {
            "background": "url(testPicture)",
            "float": "left",
            "marginLeft": "3px",
            "marginTop": "4px",
          }
        }
      />
      <span
        style={
          Object {
            "color": "#e65d78",
            "marginLeft": "15px",
            "position": "relative",
            "top": "12px",
          }
        }
      >
        testName
      </span>
    </div>
  </li>,
  "nodes": Array [
    <li
      id="testId"
      onClick={[Function]}
      style={
        Object {
          "background": "#f1f1f1",
          "borderLeft": "3px solid #e65d78",
          "boxSizing": "border-box",
          "display": "flex",
          "fontSize": 14,
          "height": "50px",
          "listStyleType": "none",
          "padding": "3px 0 3px 6px",
        }
      }
    >
      <div>
        <span
          className="lmc-profile-picture__small"
          style={
            Object {
              "background": "url(testPicture)",
              "float": "left",
              "marginLeft": "3px",
              "marginTop": "4px",
            }
          }
        />
        <span
          style={
            Object {
              "color": "#e65d78",
              "marginLeft": "15px",
              "position": "relative",
              "top": "12px",
            }
          }
        >
          testName
        </span>
      </div>
    </li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LmcResidentsSidebarItem
        isSelected={true}
        onClick={[Function]}
        resident={
          Object {
            "fields": Object {
              "picture": "testPicture",
            },
            "id": "testId",
            "name": "testName",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isSelected": true,
          "onClick": [Function],
          "resident": Object {
            "fields": Object {
              "picture": "testPicture",
            },
            "id": "testId",
            "name": "testName",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          id="testId"
          onClick={[Function]}
          style={
            Object {
              "background": "#f1f1f1",
              "borderLeft": "3px solid #e65d78",
              "boxSizing": "border-box",
              "display": "flex",
              "fontSize": 14,
              "height": "50px",
              "listStyleType": "none",
              "padding": "3px 0 3px 6px",
            }
          }
        >
          <div>
            <span
              className="lmc-profile-picture__small"
              style={
                Object {
                  "background": "url(testPicture)",
                  "float": "left",
                  "marginLeft": "3px",
                  "marginTop": "4px",
                }
              }
            />
            <span
              style={
                Object {
                  "color": "#e65d78",
                  "marginLeft": "15px",
                  "position": "relative",
                  "top": "12px",
                }
              }
            >
              testName
            </span>
          </div>
        </li>,
        "_debugID": 2,
        "_renderedOutput": <li
          id="testId"
          onClick={[Function]}
          style={
            Object {
              "background": "#f1f1f1",
              "borderLeft": "3px solid #e65d78",
              "boxSizing": "border-box",
              "display": "flex",
              "fontSize": 14,
              "height": "50px",
              "listStyleType": "none",
              "padding": "3px 0 3px 6px",
            }
          }
        >
          <div>
            <span
              className="lmc-profile-picture__small"
              style={
                Object {
                  "background": "url(testPicture)",
                  "float": "left",
                  "marginLeft": "3px",
                  "marginTop": "4px",
                }
              }
            />
            <span
              style={
                Object {
                  "color": "#e65d78",
                  "marginLeft": "15px",
                  "position": "relative",
                  "top": "12px",
                }
              }
            >
              testName
            </span>
          </div>
        </li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcResidentsSidebarItem
    isSelected={true}
    onClick={[Function]}
    resident={
      Object {
        "fields": Object {
          "picture": "testPicture",
        },
        "id": "testId",
        "name": "testName",
      }
    }
  />,
}
`;
